lib/PINNBench/Burgers/model.py

```diff
@@ -0,0 +1,329 @@
+        x_u_train = x_train[0]
+        x_f_train = x_train[1]
+
+        x_u = x_u_train[:, 0]
+        t_u = x_u_train[:, 1]
+
+        x_f = x_f_train[:, 0]
+        t_f = x_f_train[:, 1]
+        for i in range(self.epochs):
+            train_ls = []
+            # for j in range(len(x_f_train) // self.batch_size):
+            #     x_train_batch = x_u_train#[j*self.batch_size : (j+1)*self.batch_size]
+            #     y_train_batch = y_train#[j*self.batch_size : (j+1)*self.batch_size]
+            #     x_f_train_batch = x_f_train[j*self.batch_size : (j+1)*self.batch_size]
+            with tf.GradientTape() as t:
+                u_out = self.net(x_u, t_u)
+                f_out = self.f_net(x_f, t_f)
+                loss_u = self.loss(y_train, u_out)
+                loss_f = self.loss(0, f_out) 
+                tot_loss = loss_u + self.alpha * loss_f
+                train_ls.append(tot_loss)
+            grads = t.gradient(tot_loss, self.net.trainable_weights)
+            self.optimizer.apply_gradients(zip(grads, self.net.trainable_weights))
+        
+            # validation loss
+            #val_r2 = self.r2(y_val, self.net(x_val[:, 0], x_val[:, 1]))
+            val_f_loss = self.loss(0, self.f_net(x_val[:, 0], x_val[:, 1])).numpy()
+            print("Epoch %d, training loss: (%.4f, %.4f), validiation PDE loss %.4f" % (i+1, loss_u.numpy(), loss_f.numpy(), val_f_loss))
+        return val_f_loss # return the objective.
+                
+    def r2(self, true, pred):
+        true = tf.cast(true, tf.float32)
+        rss = tf.reduce_sum((true - pred)**2)  
+        tss = tf.reduce_sum((true - tf.reduce_mean(true))**2)
+        return 1 - rss/tss
+
+    def test(self, x_test, y_test):
+        out = self.net(x_test)
+        r2 = self.r2(y_test, out)
+        f_loss = self.f_loss(x_test)
+        return r2, f_loss
+
+
+
+def get_data(NT, NX, TMAX, XMAX, NU):
+    """
+    NT: number of points in time.
+    NX: number of points in space.
+    TMAX: the maximum time span.
+    XMAX: the maximum space span.
+    NU: viscosity
+    """
+    u,x = convection_diffusion(NT, NX, TMAX, XMAX, NU) # shape of [x, t]
+    x = np.linspace(0, XMAX, NX)
+    t = np.linspace(0, TMAX, NT)
+    xv, tv = np.meshgrid(x, t)
+    x_= np.stack((xv.flatten(), tv.flatten()), axis=1)
+    y_= u.T.flatten()
+
+    # Get the indices of points on the boundaries and initial conditions.
+    b_up_idx = np.where(x_[:,0]==x[-1])[0]
+    b_low_idx = np.where(x_[:,0]==x[0])[0]
+    ic_idx = np.where(x_[:,1]==t[0])[0]
+  
+    u_train_idx = np.concatenate([b_up_idx, b_low_idx, ic_idx]).reshape(-1)
+    x_u_train = x_[u_train_idx]
+    y_u_train = y_[u_train_idx]
+
+    NUM_BC_PT = 500
+    u_idx = np.random.permutation(len(x_u_train))[:NUM_BC_PT]
+    NUM_COL_PT = 2000#int(.1 * len(x_)) # 10% of the points in domain.
+    print("Generating data... total size %d, number of collocation points for trianing %d" % (len(x_), NUM_COL_PT))
```

Comment:
when using print statements in code, either use the `logging` module which allows to active/deactive printing statements and log outputs in file or in stdout, or add a `verbose=0/1` parameter to your function/objects to allow to activate/deactivate the printing.

Reply:
I added a PDE solver for the Burgers equation for a more general data generation process and removed the print line.